src/libasr/codegen/asr_to_c.cpp

```diff
@@ -529,7 +524,12 @@ class ASRToCVisitor : public BaseCCPPVisitor<ASRToCVisitor>
             } else if (ASR::is_a<ASR::List_t>(*v_m_type)) {
                 ASR::List_t* t = ASR::down_cast<ASR::List_t>(v_m_type);
                 std::string list_type_c = c_ds_api->get_list_type(t);
-                sub = format_type_c("", list_type_c, v.m_name,
+                std::string name = v.m_name;
+                if (name == "_lpython_return_variable" && v.m_intent == ASRUtils::intent_out) {
```

Comment:
This is hardwiring a check for `_lpython_return_variable` at several places in the C backend.

However, the C backend should work with any frontend, and should be independent of how the frontend names its variables.

Is there a way to fix it without any such checks of the type `name == "_lpython_return_variable"`?

Reply:
Shall look into it !

Reply:
There are a couple places to consider if you think about it 
1) Whenever use use the `subroutine_from_function`, we would end up with this ASR node 
```
        (=
            (Var 3 _lpython_return_variable)
            (Var 3 some_var)
            ()
        )
```
As we are finally storing whatever answer we want to return in the variable of intent `Out` (which in our case and I think most cases would be `_lpython_return_variable` or basically the variable to be returned.)
So I think we could do with just the `m_intent == ASRUtils::intent_out` check for any normal case as we know the `return_var` is changed to `out`. Only when some exceptional/corner case comes up we might have to think about this

The "_lpython_return_variable" just adds another level of security and in general gives more protection.

Reply:
I would say we can just use it once in the beggining (when we hit the variable)
```
  mmrv:
      (Function
          (SymbolTable
              3
              {
                  _lpython_return_variable:
                      (Variable
                          3
                          _lpython_return_variable
                          []
                          Out
                          ()
                          ()
                          Default
                          (List
                              (CPtr)
                          )
                          ()
                          Source
                          Public
                          Required
                          .false.
                      ),
  ```
  Here we can check 
  ```
                if (name == "_lpython_return_variable" && v.m_intent == ASRUtils::intent_out) {
                    is_return_var_intent_out = true;
  ```
  Once this is done we can always use the bool variable `is_return_var_intent_out` for anything we want.

Reply:
After some thought and checking for some corner cases I've realized that we might be able to do without using "_lpython_return_variable" but it's best (in terms of convinience) to use it. Let's discuss more in the call today.